### DISCUSSION TITLE
Does modV 3.x support videos?

### DISCUSSION THREAD

**Discussion by izackpaz** [vcync/modV#501] [Q&A] [answered]
ive been tinkering around with modv since there isnt really much software out there that does what im looking for and i see in the website that you can add videos on modv but i havent been able to find out how to do that,, can anyone help me?

**Comment by 2xAA** [comment, marked as answer]
Hi @izackpaz.

The latest version of modV (3.x) currently doesn't support video as we changed the architecture of the application heavily.

We're looking to bring video back in upcoming versions.

For now, one approach would be to use a virtual webcam application to play back video and use the `Webcam` module to display this.

For Windows and macOS you can use OBS as a virtual webcam:
https://streamshark.io/blog/using-obs-as-a-virtual-webcam-on-windows-and-macos/

If you're looking for something more lightweight on macOS you could also use CamTwist:
http://camtwiststudio.com/

A quick Google presents these options for Linux, however I've never used them myself:
https://github.com/webcamoid/akvcam
https://github.com/umlaeute/v4l2loopback

**Reply by izackpaz** [reply]
does an older version of modv support putting mp4s?

**Reply by 2xAA** [reply]
@izackpaz Yes, 2.0 found on the 2.0 branch supports videos.
However, we do not support modV 2.0.

We can try to help where possible but we won't push any fixes.

**Reply by TimPietrusky** [reply]
@izackpaz you can also keep track of the feature via https://github.com/vcync/modV/issues/521

**Reply by 2xAA** [reply]
Video is now supported as of 3.27.0!
https://github.com/vcync/modV/releases/tag/3.27.0

Place videos in the "video" folder in a project in the media folder - the default path would look like `/path/to/media/folder/default/video` and use the module "Texture2D" in modV.
